libs/blocks/merch-card/merch-card.js

```diff
@@ -1,184 +1,206 @@
+  if (el) {
+    const metadata = getMetadata(el);
+    styleClasses = metadata?.style?.text?.split(',').map((token) => token.split(' ').join('-')) ?? [];
+  }
+  if (!merchCardGrids.some((styleClass) => styleClasses.includes(styleClass))) {
+    section.classList.add('three-merch-cards');
+  }
+}
+
 const init = (el) => {
-  loadStyle(`${base}/deps/caas.css`);
-  const section = el.closest('.section');
-  section.classList.add('milo-card-section');
-  const images = el.querySelectorAll('picture');
-  let image;
-  const icons = [];
-  const rows = el.querySelectorAll(':scope > *');
-  const ribbonMetadata = rows[0].children?.length === 2 ? rows[0].children : null;
-  const row = rows[ribbonMetadata === null ? 0 : 1];
-  const altCta = rows[rows.length - 1].children?.length === 2
-    ? rows[rows.length - 1].children : null;
-  const allPElems = row.querySelectorAll('p');
-  const ctas = allPElems[allPElems.length - 1];
-  const styles = [...el.classList];
-  const cardType = getPodType(styles);
-  const merchCard = el;
-  decorateBlockHrs(el);
-  images.forEach((img) => {
-    const imgNode = img.querySelector('img');
-    const { width, height } = imgNode;
-    const isSquare = Math.abs(width - height) <= 10;
-    if (img) {
-      if (isSquare) {
-        icons.push(img);
-      } else {
-        image = img;
+  let section = el.closest('.section');
+  const upClass = getUpFromSectionMetadata(section);
+  if (upClass) {
+    import('./legacy-merch-card.js')
+      .then((module) => {
+        module.initLegacyMerchCard(el);
+      });
+  } else {
+    if (section.parentElement.classList.contains('fragment')) {
+      const fragment = section.parentElement;
+      const fragmentParent = fragment.parentElement;
+      section.style.display = 'contents';
+      fragment.style.display = 'contents';
+      fragmentParent.style.display = 'contents';
+      section = fragmentParent.parentElement;
+    }
+    addMerchCardGridsIfMissing(section);
+    const headings = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
+    decorateLinkAnalytics(el, headings);
+    const images = el.querySelectorAll('picture');
+    let image;
+    const icons = [];
+    const rows = el.querySelectorAll(':scope > *');
+    const styles = [...el.classList];
+    const cardType = getPodType(styles);
+    const ribbonMetadata = rows[0].children?.length === 2 ? rows[0].children : null;
+    const actionMenuContent = cardType === 'catalog' ? getActionMenuContent(el, ribbonMetadata) : null;
+    const row = getMerchCardRows(rows, ribbonMetadata, cardType, actionMenuContent);
+    const altCta = rows[rows.length - 1].children?.length === 2
+      ? rows[rows.length - 1].children : null;
+    const allPElems = row.querySelectorAll('p');
+    const ctas = allPElems[allPElems.length - 1];
+    images.forEach((img) => {
+      const imgNode = img.querySelector('img');
+      const { width, height } = imgNode;
+      const isSquare = Math.abs(width - height) <= 10;
+      if (img) {
```

Comment:
Maybe move this check up? if `img` is null or not defined, wouldn't `const imgNode = img.querySelector('img');` fail?

Reply:
Since it's inside a forEach `imgNode` only initializes if the array `images` has items.